<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>828</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <widget class="QSplitter" name="vsplitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QSplitter" name="splitter">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <widget class="Viewer" name="m_viewport">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>1</width>
          <height>1</height>
         </size>
        </property>
       </widget>
       <widget class="QTabWidget" name="tabWidget">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>1</width>
          <height>1</height>
         </size>
        </property>
        <property name="tabPosition">
         <enum>QTabWidget::North</enum>
        </property>
        <property name="currentIndex">
         <number>3</number>
        </property>
        <property name="movable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="m_object_tab">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <attribute name="title">
          <string>Object</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QGroupBox" name="groupBox_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Transform</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="spacing">
              <number>0</number>
             </property>
             <item>
              <layout class="QGridLayout" name="gridLayout">
               <property name="verticalSpacing">
                <number>0</number>
               </property>
               <item row="1" column="3">
                <widget class="QDoubleSpinBox" name="m_scale_x">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="prefix">
                  <string/>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QDoubleSpinBox" name="m_move_x">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="prefix">
                  <string/>
                 </property>
                 <property name="minimum">
                  <double>-99999.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="value">
                  <double>0.000000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QDoubleSpinBox" name="m_move_z">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="prefix">
                  <string/>
                 </property>
                 <property name="minimum">
                  <double>-99999.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QDoubleSpinBox" name="m_move_y">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="prefix">
                  <string/>
                 </property>
                 <property name="minimum">
                  <double>-99999.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <widget class="QDoubleSpinBox" name="m_rotate_x">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="prefix">
                  <string/>
                 </property>
                </widget>
               </item>
               <item row="2" column="2">
                <widget class="QDoubleSpinBox" name="m_rotate_y">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="prefix">
                  <string/>
                 </property>
                </widget>
               </item>
               <item row="3" column="2">
                <widget class="QDoubleSpinBox" name="m_rotate_z">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="prefix">
                  <string/>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>Rotate</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>Translate</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="3">
                <widget class="QDoubleSpinBox" name="m_scale_z">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="prefix">
                  <string/>
                 </property>
                </widget>
               </item>
               <item row="0" column="3">
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>Scale</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="3">
                <widget class="QDoubleSpinBox" name="m_scale_y">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="prefix">
                  <string/>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_4">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>X</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_5">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Y</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="label_6">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Z</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Drawing</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="QCheckBox" name="m_draw_bbox">
               <property name="text">
                <string>Bounding Box</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="m_draw_tree">
               <property name="text">
                <string>Topology</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_4">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Properties</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QLabel" name="label_10">
                 <property name="text">
                  <string>Voxel Size</string>
                 </property>
                 <property name="buddy">
                  <cstring>m_voxel_size</cstring>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="m_voxel_size">
                 <property name="minimum">
                  <double>0.010000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.010000000000000</double>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="m_scene_tab">
         <attribute name="title">
          <string>Scene</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QPushButton" name="m_start_but">
            <property name="text">
             <string>Play Animation</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Frames Per Second</string>
              </property>
              <property name="buddy">
               <cstring>m_fps</cstring>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="m_fps">
              <property name="value">
               <double>24.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Start Frame</string>
              </property>
              <property name="buddy">
               <cstring>m_star_frame</cstring>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>End Frame</string>
              </property>
              <property name="buddy">
               <cstring>m_end_frame</cstring>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="m_star_frame"/>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="m_end_frame">
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="value">
               <number>250</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>World</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="title">
             <string>Display</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QPushButton" name="pushButton">
               <property name="text">
                <string>Change Color</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="m_draw_grid">
               <property name="text">
                <string>Draw Grid</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Sculpt</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QGroupBox" name="groupBox_5">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Brush Settings</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="1" column="0">
              <widget class="QLabel" name="label_12">
               <property name="text">
                <string>Radius</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QComboBox" name="m_brush_mode">
               <item>
                <property name="text">
                 <string>Add</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Subtract</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_11">
               <property name="text">
                <string>Mode</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QDoubleSpinBox" name="m_brush_radius">
               <property name="value">
                <double>5.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_13">
               <property name="text">
                <string>Amount</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QDoubleSpinBox" name="m_brush_amount">
               <property name="singleStep">
                <double>0.010000000000000</double>
               </property>
               <property name="value">
                <double>0.100000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </widget>
      <widget class="TimeLineWidget" name="m_timeline" native="true"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>828</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFichier">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="m_open_file_act"/>
   </widget>
   <widget class="QMenu" name="menuAdd">
    <property name="title">
     <string>Add</string>
    </property>
    <addaction name="m_add_cube"/>
    <addaction name="actionLevel_Set_Sphere"/>
   </widget>
   <addaction name="menuFichier"/>
   <addaction name="menuAdd"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="m_open_file_act">
   <property name="text">
    <string>Open VDB File</string>
   </property>
  </action>
  <action name="m_add_cube">
   <property name="text">
    <string>Cube</string>
   </property>
  </action>
  <action name="actionLevel_Set_Sphere">
   <property name="text">
    <string>Level Set</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TimeLineWidget</class>
   <extends>QWidget</extends>
   <header>timelinewidget.h</header>
   <container>1</container>
   <slots>
    <slot>setStartFrame(int)</slot>
    <slot>setEndFrame(int)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>Viewer</class>
   <extends>QOpenGLWidget</extends>
   <header>viewer.h</header>
   <slots>
    <slot>changeBackground()</slot>
    <slot>drawGrid(bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>m_move_x</tabstop>
  <tabstop>m_move_y</tabstop>
  <tabstop>m_move_z</tabstop>
  <tabstop>m_rotate_x</tabstop>
  <tabstop>m_rotate_y</tabstop>
  <tabstop>m_rotate_z</tabstop>
  <tabstop>m_scale_x</tabstop>
  <tabstop>m_scale_y</tabstop>
  <tabstop>m_scale_z</tabstop>
  <tabstop>m_draw_bbox</tabstop>
  <tabstop>m_draw_tree</tabstop>
  <tabstop>m_voxel_size</tabstop>
  <tabstop>m_start_but</tabstop>
  <tabstop>m_fps</tabstop>
  <tabstop>m_star_frame</tabstop>
  <tabstop>m_end_frame</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>m_draw_grid</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>m_open_file_act</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_add_cube</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>addCube()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>413</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLevel_Set_Sphere</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>addLevelSet()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>413</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_draw_tree</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>795</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>410</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_draw_bbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateObject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>795</x>
     <y>270</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>m_viewport</receiver>
   <slot>changeBackground()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>745</x>
     <y>116</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_draw_grid</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_viewport</receiver>
   <slot>drawGrid(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>745</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_start_but</sender>
   <signal>pressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>startAnimation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>493</x>
     <y>81</y>
    </hint>
    <hint type="destinationlabel">
     <x>413</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_end_frame</sender>
   <signal>valueChanged(int)</signal>
   <receiver>m_timeline</receiver>
   <slot>setEndFrame(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>806</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>568</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_star_frame</sender>
   <signal>valueChanged(int)</signal>
   <receiver>m_timeline</receiver>
   <slot>setStartFrame(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>806</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>568</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openFile()</slot>
  <slot>updateObject()</slot>
  <slot>addCube()</slot>
  <slot>addLevelSet()</slot>
  <slot>startAnimation()</slot>
 </slots>
</ui>
